client/client.go

```diff
@@ -48,6 +52,10 @@ type Config struct {
 	// BaseURL contains the base URL where snappy daemon is expected to be.
 	// It can be empty for a default behavior of talking over a unix socket.
 	BaseURL string
+
+	// Public determines whether or not this client will talk to the private or
+	// public snapd socket (private by default).
+	Public bool
```

Comment:
I've actually had to revert that commit. Using `os.Stat()` doesn't seem to work. For reasons I don't understand it says the socket is perfectly readable, even though connecting to it causes the process to hang. More information in the reverting commit.